### DISCUSSION TITLE
Open Grid in new Dialog Window - Works first time then says it exists on second time.

### DISCUSSION THREAD

**Discussion by stixoffire** [serenity-is/Serenity#6456] [Help] [answered]
I have a dialog which I open with 
dlg = new mydialog)_;
dlg.Initialize(); 
dlg.template = "<div id='trgrd'>
dlg.open();

In my opened dialog I create the grid in the event for onDialogOpen().
grid = myGrid(#'trgrd');

while my dialog opens and the grid is displayed and functions and I can save data - this works on the first time only but the second time I open a record and then click my button for the new dialog it errors with a message that MyGrid already exists in the dom.

I think I am doing something wrong when creating the grid ? 

I will try to post my code from the other computer but right now this is all I can post.

**Comment by haajit** [comment]
You should create grid in constructor of dialog

**Comment by stixoffire** [comment, marked as answer]
I needed to create my template with the ~_TrGrid andpass that in the Grid("TrGrid") that worked with out issue.
